spec/e2e/query_proxy_methods_spec.rb

```diff
@@ -72,6 +72,110 @@ def destroy_called
     end
   end
 
+  describe 'find_or_initialize_by' do
+    context 'with relations chain' do
+      let(:emily)       { Student.create(name: 'Emily') }
+      let(:philosophy)  { Lesson.create(name: 'philosophy') }
+      before do
+        philosophy.students << jimmy
+      end
+
+      it 'returns the matching node when exists' do
+        expect(philosophy.students.find_or_initialize_by(name: jimmy.name).name).to eq(jimmy.name)
+      end
+
+      it 'initializes and associates a new node when it didn\'t exist' do
+        expect(philosophy.students.where(name: 'Rebecca').first).to be_nil
+        student = philosophy.students.find_or_initialize_by(name: 'Rebecca')
+        expect(student).not_to be_persisted
+        expect(student.name).to eq('Rebecca')
+        expect(student.lessons.to_a).to include(philosophy)
+        expect { student.save }.to change { philosophy.reload.students.count }.by(1)
+      end
+
+      it 'returns the node after initializing' do
+        expect(philosophy.students.find_or_initialize_by(name: 'Jacob')).to be_a(Neo4j::ActiveNode)
```

Comment:
Since all of these are doing a `find_or_initialize` by on `philosophy.students` perhaps you could DRY it up a bit like:

```ruby
describe 'find_or_initialize_by on an association' do
  subject { philosophy.students.find_or_initialize_by(name: 'Jacob') }

  it { should be_a(Neo4j::ActiveNode) }
  its(:name) { should eq('Jacob') }
  it { should_not be_persisted }
  its(:lessons) { should include(philosophy) }
  it 'can be saved' do
    expect { subject.save }.to change { philosophy.reload.students.count }.by(1)
  end
end
```